lib/tasks/CRM457_2154.rake

```diff
@@ -0,0 +1,47 @@
+# https://dsdmoj.atlassian.net/browse/CRM457-2154
+
+namespace :CRM457_2154 do
+  desc "Append high_value flag to submissions"
+  task adds_high_value: :environment do
+    # Query to select all submissions that don't have a high_value flag
+    # and go through each submission version to add the high_value flag:
+    # use cost_summary's gross_cost if present, otherwise use submission's application risk
+    submissions = Submission.where(application_type: 'crm7')
+                            .joins(:ordered_submission_versions)
```

Comment:
You could simplify this logic slightly if you switched it round:

versions = SubmissionVersion.includes(:submission).joins(:submission).where("application_version.application -> 'cost_summary' ->> 'high_value' IS NULL AND application.application_type = 'CRM7'"

That way you avoid a nested `each` loop and do fewer DB lookups